gtda/homology/simplicial.py

```diff
@@ -532,17 +529,253 @@ def plot(Xt, sample=0, homology_dimensions=None, plotly_params=None):
+
+    See also
+    --------
+    VietorisRipsPersistence, FlagserPersistence, SparseRipsPersistence, \
+    EuclideanCechPersistence
+
+    Notes
+    -----
+    Delaunay triangulation are computed by :class:`scipy.spatial.Delaunay`.
+    Ripser <https://github.com/Ripser/ripser>`_ is used as a C++ backend for
+    computing Vietoris–Rips persistent homology. Python bindings were modified
+    for performance from the `ripser.py
+    <https://github.com/scikit-tda/ripser.py>`_ package.
+
+    Persistence diagrams produced by this class must be interpreted with
+    care due to the presence of padding triples which carry no information.
+    See :meth:`transform` for additional information.
+
+    References
+    ----------
+    [1] U. Bauer, "Ripser: efficient computation of Vietoris–Rips persistence \
+        barcodes", 2019; `arXiv:1908.02518 \
+        <https://arxiv.org/abs/1908.02518>`_.
+
+    """
+
+    _hyperparameters = {
+        'homology_dimensions': {
+            'type': (list, tuple),
+            'of': {'type': int, 'in': Interval(0, np.inf, closed='left')}
+            },
+        'coeff': {'type': int, 'in': Interval(2, np.inf, closed='left')},
+        'max_edge_length': {'type': Real},
+        'infinity_values': {'type': (Real, type(None))}
+        }
+
+    def __init__(self, homology_dimensions=(0, 1), coeff=2,
+                 max_edge_length=np.inf, infinity_values=None, n_jobs=None):
+        self.homology_dimensions = homology_dimensions
+        self.coeff = coeff
+        self.max_edge_length = max_edge_length
+        self.infinity_values = infinity_values
+        self.n_jobs = n_jobs
+
+    def _weak_alpha_diagram(self, X):
+        # `indices` will serve as the array of column indices
+        indptr, indices = Delaunay(X).vertex_neighbor_vertices
+
+        # Compute the array of row indices
+        row = np.zeros_like(indices)
+        row[indptr[1:-1]] = 1
+        np.cumsum(row, out=row)
+
+        # We only need the upper diagonal
+        mask = indices > row
+        row, col = row[mask], indices[mask]
+        dists = np.linalg.norm(X[row] - X[col], axis=1)
+        # Note: passing the shape explicitly should not be needed in more
+        # recent versions of C++ ripser
+        n_points = len(X)
+        dm = coo_matrix((dists, (row, col)), shape=(n_points, n_points))
+
+        Xdgms = ripser(dm, maxdim=self._max_homology_dimension,
+                       thresh=self.max_edge_length, coeff=self.coeff,
+                       metric='precomputed')['dgms']
+
+        if 0 in self._homology_dimensions:
+            Xdgms[0] = Xdgms[0][:-1, :]  # Remove one infinite bar
+
+        return Xdgms
+
+    def fit(self, X, y=None):
+        """Calculate :attr:`infinity_values_`. Then, return the estimator.
+
+        This method is here to implement the usual scikit-learn API and hence
+        work in pipelines.
+
+        Parameters
+        ----------
+        X : ndarray or list
```

Comment:
It seems to me information about allowed types should be more complete and the description should stress the point of having a list as an input. Something like:

```
X :  list of n_samples 2D ndarrays or ndarray of shape (n_samples, n_points, n_dimensions)
Input data representing a collection of point clouds. If it is a list, all point clouds can live in spaces of different dimensions.
```

Reply:
This text is currently consistent between all other transformers. I don't think I agree about overburdening the text with shapes at the top, as shapes are explained in complete detail in the description. But maybe there can be a middle way, such as
```
X :  ndarray or list of length n_samples
```
But I agree about remarking on different dimensions. I'll apply this consistently in simplicial.py.

Reply:
I also agree that the conventions on the shapes of point clouds (is it (n_points, n_dimensions) or the reverse) are not explained and they should be, I've pushed changes in which they are. Let me know!

Reply:
I think that's perfectly acceptable.